sklearn/datasets/openml.py

```diff
@@ -45,13 +46,35 @@ def _open_openml_url(openml_path, data_home):
         Directory to which the files will be cached. If None, no caching will
         be applied.
 
+    md5_checksum: str
+        Checksum of the OpenML ressource downloaded. If None, no verification
+        is done.
+
     Returns
     -------
     result : stream
         A stream to the OpenML resource
     """
     if data_home is None:
-        return urlopen(_OPENML_PREFIX + openml_path)
+
+        response = urlopen(_OPENML_PREFIX + openml_path)
+
+        if md5_checksum is None:
+            return response
+
+        stream = io.BytesIO()
```

Comment:
there seems to be some duplication between here and a few lines below. Could you factor that out in a function or something?